src/timed-event-manager.ts

```diff
@@ -398,99 +408,75 @@ export class TimedEventManager {
   }
 
   /**
-   * Check whether the current event is active and for April Fools.
+   * Check whether the current {@linkcode TimedEvent} is active and for April Fools.
    * @returns Whether the April Fools event is currently active.
    */
   isAprilFoolsActive(): boolean {
-    return timedEvents.some(
-      te => this.isActive(te) && te.hasOwnProperty("bannerKey") && te.bannerKey!.startsWith("aprf"),
-    );
+    return this.activeEvent()?.bannerKey?.startsWith("aprf") ?? false;
   }
 
   activeEventHasBanner(): boolean {
-    const activeEvents = timedEvents.filter(te => this.isActive(te) && te.hasOwnProperty("bannerKey"));
-    return activeEvents.length > 0;
+    return this.activeEvent()?.bannerKey != null;
   }
 
-  getShinyMultiplier(): number {
-    let multiplier = 1;
-    const shinyEvents = timedEvents.filter(te => te.eventType === EventType.SHINY && this.isActive(te));
-    for (const se of shinyEvents) {
-      multiplier *= se.shinyMultiplier ?? 1;
-    }
+  /**
+   * Get the multiplier for shiny encounters during a shiny {@linkcode TimedEvent}
+   * @returns the shiny encounter multiplier
+   */
+  getShinyEncounterMultiplier(): number {
+    return this.activeEvent()?.shinyEncounterMultiplier ?? 1;
+  }
 
-    return multiplier;
+  /**
+   * Get the multiplier for shiny catches during a shiny {@linkcode TimedEvent}
+   * @returns the shiny catch multiplier
+   */
+  getShinyCatchMultiplier(): number {
+    return this.activeEvent()?.shinyCatchMultiplier ?? 1; // todo: add base constant from #6670 after it gets merged
```

Comment:
I have a feeling we'll need to do that ourselves.
```suggestion
    return this.activeEvent()?.shinyCatchMultiplier ?? 1;
```